File: javascript/control_flow/switch_statement/switch_statement.js
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

let now = "afternoon";

switch (now)
{
  case "morning":
    console.log("Good morning!");
    break;
  case "afternoon":
    console.log("Good afternoon!");
    break;
  case "evening":
    console.log("Good evening!");
    break;
  case "night":
    console.log("Good evening!");
    break;
}

now = "evening";

switch (now)
{
  case "morning":
    console.log("Good morning!");
    break;
  case "afternoon":
    console.log("Good afternoon!");
    break;
  case "evening":
  case "night":
    console.log("Good evening!");
    break;
}

now = "other";

switch (now)
{
  case "morning":
    console.log("Good morning!");
    break;
  case "afternoon":
    console.log("Good afternoon!");
    break;
  case "evening":
    console.log("Good evening!");
    break;
  case "night":
    console.log("Good evening!");
    break;
  default:
    console.log("Hello!");
    break;
}

File: c#/control_flow/switch_statement/SwitchStatement.cs
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

class SwitchStatement {
  enum PartsOfTheDay {
    morning,
    afternoon,
    evening,
    night,
    other
  };

  public static void Main(string[] args) {
    PartsOfTheDay now;

    now = PartsOfTheDay.afternoon;

    switch (now)
    {
      case PartsOfTheDay.morning:
        System.Console.WriteLine("Good morning!");
        break;
      case PartsOfTheDay.afternoon:
        System.Console.WriteLine("Good afternoon!");
        break;
      case PartsOfTheDay.evening:
        System.Console.WriteLine("Good evening!");
        break;
      case PartsOfTheDay.night:
        System.Console.WriteLine("Good evening!");
        break;
    }

    now = PartsOfTheDay.evening;

    switch (now)
    {
      case PartsOfTheDay.morning:
        System.Console.WriteLine("Good morning!");
        break;
      case PartsOfTheDay.afternoon:
        System.Console.WriteLine("Good afternoon!");
        break;
      case PartsOfTheDay.evening:
      case PartsOfTheDay.night:
        System.Console.WriteLine("Good evening!");
        break;
    }

    now = PartsOfTheDay.other;

    switch (now)
    {
      case PartsOfTheDay.morning:
        System.Console.WriteLine("Good morning!");
        break;
      case PartsOfTheDay.afternoon:
        System.Console.WriteLine("Good afternoon!");
        break;
      case PartsOfTheDay.evening:
        System.Console.WriteLine("Good evening!");
        break;
      case PartsOfTheDay.night:
        System.Console.WriteLine("Good evening!");
        break;
      default:
        System.Console.WriteLine("Hello!");
        break;
    }
  }
}
